pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -632,5 +636,128 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
 
         applyMatrix(matrix.data(), wires, inverse);
     }
+
+    /**
+     * @brief Apply a mid-circuit measurement.
+     *
+     * @param wires Wires to sample from.
+     * @param postselect Value to postselect (0 or 1) or empty to ignore
+     * @param reset Flag to reset wire to 0
+     */
+    inline auto
+    applyMidMeasureMP(const std::vector<std::size_t> &wires,
+                      const std::vector<std::size_t> &postselect = {},
+                      bool reset = false) -> int {
+        PL_ABORT_IF_NOT(wires.size() == 1,
+                        "MidMeasureMP should have a single wire.")
+        PL_ABORT_IF(postselect.size() > 1,
+                    "MidMeasureMP accepts at most one postselect value.")
+        const int ps = postselect.empty() ? -1 : postselect[0];
+
+        std::vector<PrecisionT> probs_ = probs(wires[0]);
+        auto sample = randomSample(probs_[0]);
+        if (ps != -1 && ps != sample) {
+            sample = -1;
+        } else {
+            collapse(wires[0], sample);
+            if (reset && sample == 1) {
+                applyOperation("PauliX", wires);
+            }
+        }
+        return sample;
+    }
```

Comment:
This class is realizing a measurement. It would be nice to have it in the measurement class. Without looking too deep into that, it looks like this will integrate nicely with the class design we have in place for LightningQubit with the new device API.

Reply:
ok, I'll move `applyMidMeasureMP` to Measurement

Reply:
As agreed upon, `applyMidMeasureMP` has been moved to the python layer